app/services/adjustment_create_service.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class AdjustmentCreateService
+  include ServiceObjectErrorsMixin
+  attr_reader :adjustment
+
+  def initialize(adjustment_params)
+    @adjustment = Adjustment.new(adjustment_params)
+  end
+
+  def call
+    # Combine line items for adjustment
+    combine_adjustment
+    # Check for validity, and save the actual adjustment
+
+    if @adjustment.valid? && enough_inventory_for_decreases?
+      ActiveRecord::Base.transaction do
+        # Make the necessary changes in the db
+        @adjustment.save
+        # Split into positive and negative portions.  NOTE -- THIS CHANGES THE ORIGINAL LINE ITEMS DO **NOT** RESAVE
```

Comment:
What does this comment mean? What should we not resave?

Reply:
REvised comment coming your way RSN